youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,243 @@
+            'title': 'Dentro la notizia - Ferrari, cosa succede a Maranello',
+            'description': 'md5:9907d65b53765681fa3a0b3122617c1f',
+            'thumbnail': r're:^https://www\.repstatic\.it/video/photo/.+?-thumb-social-play\.jpg$',
+        },
+    }, {
+        'url': 'https://video.espresso.repubblica.it/embed/tutti-i-video/01-ted-villa/14772/14870&width=640&height=360',
+        'md5': 'ca3323b47c94cac92fff03eef0387d97',
+        'info_dict': {
+            'id': '14772/14870',
+            'ext': 'mp4',
+            'title': 'Festival EMERGENCY, Villa: «La buona informazione aiuta la salute» (14772-14870)',
+            'description': 'md5:2bce954d278248f3c950be355b7c2226',
+            'thumbnail': r're:^https://www\.repstatic\.it/video/photo/.+?-thumb-social-play\.jpg$',
+        },
+    }, {
+        'url': 'https://video.messaggeroveneto.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.ilpiccolo.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.gazzettadimantova.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.mattinopadova.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.laprovinciapavese.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.tribunatreviso.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.nuovavenezia.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.gazzettadimodena.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.lanuovaferrara.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.corrierealpi.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.lasentinella.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def _clean_audio_fmts(formats):
+        # remove duplicates audio formats
+        unique_formats = []
+        for f in formats:
+            if 'acodec' in f:
+                unique_formats.append(f)
+        formats[:] = unique_formats
```

Comment:
i'm not sure why you're adding this step.

Reply:
_extract_m3u8_formats returns for some reason multiple audio url that are in fact the same. so I remove the duplicates

without cleaning 
```
format code                       extension  resolution note
audio-hls-1@48000-64000-Audio_64  m4a        audio only
audio-hls-1@48000-80000-Audio_80  m4a        audio only
audio-hls-2@48000-96000-Audio_96  m4a        audio only
audio-hls-100                     m4a        audio only  100k , mp4a.40.2
audio-hls-116                     m4a        audio only  116k , mp4a.40.2
```

cleaned
```
format code                       extension  resolution note
audio-hls-100                     m4a        audio only  100k , mp4a.40.2
audio-hls-116                     m4a        audio only  116k , mp4a.40.2
```

Reply:
they have diffirent bitrate.

Reply:
64000 is identical to hls-100
80000 is identical to hls-116

the only one different is 96000.

I'll improve the function.

Reply:
i think you should not remove the format, it's possible that they would be served from different server, protocol, etc...

Reply:
ok, Ill keep everything. The format list will be long and a bit bloated but I understand your resoning.

Reply:
I have to rectify what I said in the previos comment as I checked again.

the function _extract_m3u8_formats extracts 6 audio formats but they are in reality 3 duplicate.

Like said before:
64000 is identical to hls-100
80000 is identical to hls-116
and 96000 is identical to hls-133

here is the m3u8 manifest:
```
#EXTM3U
#EXT-X-MEDIA:TYPE=AUDIO,URI="_64/index.m3u8",GROUP-ID="1@48000-64000",NAME="Audio 64",DEFAULT=NO,AUTOSELECT=NO
#EXT-X-MEDIA:TYPE=AUDIO,URI="_80/index.m3u8",GROUP-ID="1@48000-80000",NAME="Audio 80",DEFAULT=NO,AUTOSELECT=NO
#EXT-X-MEDIA:TYPE=AUDIO,URI="_96/index.m3u8",GROUP-ID="2@48000-96000",NAME="Audio 96",DEFAULT=NO,AUTOSELECT=NO
#EXT-X-STREAM-INF:BANDWIDTH=133336,CODECS="mp4a.40.2",AUDIO="2@48000-96000"
_96/index.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=100641,CODECS="mp4a.40.2",AUDIO="1@48000-64000"
_64/index.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=116989,CODECS="mp4a.40.2",AUDIO="1@48000-80000"
_80/index.m3u8
```
as you can see there are only 3 formats but the _extract_m3u8_formats outputs 6 formats for some reason.

Reply:
then you should use `_remove_duplicate_formats` method.

Reply:
I already tried it before writing my own method (which is a modified version of that one)

The "problem" of `_remove_duplicate_formats` is that it cheched the formats in order and therefore eliminated the "good" formats and kept the "bad" ones.
By "good" I mean:
```
audio-hls-100                     m4a        audio only  100k , mp4a.40.2
```
with a clean `name`, `bitrate` and `audio codec` and not:
```
audio-hls-1@48000-64000-Audio_64  m4a        audio only
```
with a rubbish name and without bitrate and audio codec.

Because they are listed after they get eliminated... so I written a coustom method to clean the formats in a way that I liked more.

Reply:
but it would always remove formats without `acodec` even if they are only present as `EXT-X-MEDIA` formats.

Reply:
ok...

btw, I've already removed it ia couple of commits ago https://github.com/ytdl-org/youtube-dl/pull/26946/commits/8f564a71d1669eaee696fc379eab52fbb712ac30